app/controllers/returnFrequency/ConfirmVatDatesController.scala

```diff
@@ -44,15 +44,15 @@ class ConfirmVatDatesController @Inject()(val authenticate: AuthPredicate,
                                           implicit val messagesApi: MessagesApi) extends FrontendController with I18nSupport {
 
   val show: Action[AnyContent] = (authenticate andThen pendingReturnFrequency) { implicit user =>
-    user.session.get(SessionKeys.NEW_RETURN_FREQUENCY).fold {
-      Logger.info("[ConfirmVatDatesController][show] No NEW_RETURN_FREQUENCY found in session. Redirecting to Choose Dates page")
-      Redirect(controllers.returnFrequency.routes.ChooseDatesController.show().url)
-    } { newReturnFrequency =>
-      ReturnPeriod(newReturnFrequency) match {
-        case Some(newFrequency) =>
-          Ok(views.html.returnFrequency.confirm_dates(newFrequency))
-        case None => serviceErrorHandler.showInternalServerError
+    (user.session.get(SessionKeys.NEW_RETURN_FREQUENCY), user.session.get(SessionKeys.CURRENT_RETURN_FREQUENCY)) match {
+      case (Some(newFrequency), Some(currentFrequency)) => (ReturnPeriod(newFrequency), ReturnPeriod(currentFrequency)) match {
+        case (Some(nf), Some(cf)) =>
+          Ok(views.html.returnFrequency.confirm_dates(nf, cf.id.equals("Annually")))
```

Comment:
If you do the following, it avoids having "Annually" written as a String and uses the 'Annually' case object
```suggestion
          Ok(views.html.returnFrequency.confirm_dates(nf, cf == Annually))
```

Reply:
updated